test/model/content_test.dart

```diff
@@ -292,19 +293,54 @@ void main() {
       QuotationNode([ParagraphNode(links: null, nodes: [TextNode('words')])]),
     ]);
 
-  testParse('parse code blocks, no language',
+  testParse('parse code blocks, without syntax highlighting',
     // "```\nverb\natim\n```"
     '<div class="codehilite"><pre><span></span><code>verb\natim\n</code></pre></div>', const [
-      CodeBlockNode(text: 'verb\natim'),
+      CodeBlockNode([
+        CodeBlockSpanNode(text: 'verb\natim', type: CodeBlockSpanType.text),
+      ]),
     ]);
 
-  testParse('parse code blocks, with highlighted language',
+  testParse('parse code blocks, with syntax highlighting',
     // "```dart\nclass A {}\n```"
     '<div class="codehilite" data-code-language="Dart"><pre>'
         '<span></span><code><span class="kd">class</span><span class="w"> </span>'
         '<span class="nc">A</span><span class="w"> </span><span class="p">{}</span>'
         '\n</code></pre></div>', const [
-      CodeBlockNode(text: 'class A {}'),
+      CodeBlockNode([
+        CodeBlockSpanNode(text: 'class', type: CodeBlockSpanType.keywordDeclaration),
+        CodeBlockSpanNode(text: ' ', type: CodeBlockSpanType.whitespace),
+        CodeBlockSpanNode(text: 'A', type: CodeBlockSpanType.nameClass),
+        CodeBlockSpanNode(text: ' ', type: CodeBlockSpanType.whitespace),
+        CodeBlockSpanNode(text: '{}', type: CodeBlockSpanType.punctuation),
+      ]),
+    ]);
+
+  testParse('parse code blocks, with syntax highlighting and highlighted lines',
+    // "```\n::markdown hl_lines="2 4"\n# he\n## llo\n### world\n```"
+    '<div class="codehilite"><pre>'
+        '<span></span><code>::markdown hl_lines=&quot;2 4&quot;\n'
+        '<span class="hll"><span class="gh"># he</span>\n'
+        '</span><span class="gu">## llo</span>\n'
+        '<span class="hll"><span class="gu">### world</span>\n'
+        '</span></code></pre></div>', [
+      // TODO: Fix this, see comment under `CodeBlockSpanType.highlightedLines` case in lib/model/content.dart.
+      blockUnimplemented('<div class="codehilite"><pre>'
+        '<span></span><code>::markdown hl_lines=&quot;2 4&quot;\n'
+        '<span class="hll"><span class="gh"># he</span>\n'
+        '</span><span class="gu">## llo</span>\n'
+        '<span class="hll"><span class="gu">### world</span>\n'
+        '</span></code></pre></div>'),
+    ]);
+
+  testParse('parse code blocks, unknown span type',
+    // A code block span with unknown className.
```

Comment:
```suggestion
  testParse('parse code blocks, unknown span type',
    // (no markdown; this test is for future Pygments versions adding new token types)
```

In particular that makes it explicit that this test intentionally doesn't have Markdown (that that's not just a mistake), and explains why.